File: Perl/testGUI.pl
#!/usr/bin/perl

use Gtk2;
use Encode qw/ encode decode /;  #加载encode模块，用来对中文编码

Gtk2 -> init;  #Gtk+程序初始化
my $win = Gtk2::Window -> new ( );  #建立一个新的窗口
$win -> set_title ( 'Hello world program' );  #设置窗口的标题
$win -> set_size_request ( 320,240 );  #设置窗口的大小为320 240
$win -> signal_connect ( destroy => sub { Gtk2 -> main_quit; } );  #设置程序退出信号
my $word = "Hello world\n".decode ( "euc-cn",'This is my first gui to use gtk!' );  
#在屏幕几的中央写的文字，中文需要utf8编码
my $label = Gtk2::Label -> new ($word);   #新建一个标签组件
$win -> add ($label);  #在窗口中添加这个标签组件
$win -> show_all;  #显示窗口中的所有元素
Gtk2 -> main;  #开始主循环

File: Python/testGUI.py
 #!/usr/bin/python

from gi.repository import Gtk

"""
import tkinter as tk

class Application(tk.Frame):
    def __init__(self, master=None):
        tk.Frame.__init__(self, master)
        self.pack()
        self.createWidgets()

    def createWidgets(self):
        self.hi_there = tk.Button(self)
        self.hi_there["text"] = "Hello World\n(click me)"
        # bind a handler
        self.hi_there["command"] = self.say_hi
        self.hi_there.pack(side="top")

        self.QUIT = tk.Button(self, text="QUIT", fg="red",
                                            command=root.destroy)
        self.QUIT.pack(side="bottom")

    def say_hi(self):
        print("hi there, everyone!")

root = tk.Tk()
app = Application(master=root)
app.mainloop()
"""

"""
class MyWindow(Gtk.Window):

    def __init__(self):
        Gtk.Window.__init__(self, title="Hello World")

        self.button = Gtk.Button(label="Click Here")
        self.button.connect("clicked", self.on_button_clicked)
        self.add(self.button)

    def on_button_clicked(self, widget):
        print("Hello World")

win = MyWindow()
win.connect("delete-event", Gtk.main_quit)
win.show_all()
Gtk.main()
"""

class Handler:
    def onDeleteWindow(self, *args):
        Gtk.main_quit(*args)

    def onButtonClicked(self, button):
        print("Hello World!")

builder = Gtk.Builder()
builder.add_from_file("./GtkProjects/test1.glade")
builder.connect_signals(Handler())

window = builder.get_object("window1")
window.show_all()

# Start the GTK+ to process messages until the app is closed.
Gtk.main()

